metrics-operator/controllers/common/providers/prometheus/prometheus.go

```diff
@@ -149,3 +177,29 @@ func getResultForVector(result model.Value, r *KeptnPrometheusProvider) (string,
 	}
 	return value, b, nil
 }
+
+func getResultForStepMatrix(result model.Value) ([]string, []byte, error) {
+	// check if we can cast the result to a matrix
+	resultMatrix, ok := result.(model.Matrix)
+	if !ok {
+		return nil, nil, fmt.Errorf("could not cast result")
+	}
+
+	if len(resultMatrix) == 0 {
+		return nil, nil, fmt.Errorf("no values in query result")
```

Comment:
this is optional and can also be done in a follow up, but we could define common errors for could not case/no values in query result/too many values, as those are used in this function, as well as in `getResultForVector`, i.e. at the beginning of this file we would have:

```
var errCouldNotCast = fmt.Errorf("could not cast result")
var errNoValues = fmt.Errorf("no values in query result")
var errTooManyValues = fmt.Errorf("too many values in query result")
```

and use them within the two mentioned functions

Reply:
Fixed